lib/galaxy/config/__init__.py

```diff
@@ -860,10 +927,10 @@ def check(self):
             self._ensure_directory(path)
         # Check that required files exist
         tool_configs = self.tool_configs
-        if self.migrated_tools_config not in tool_configs:
+        if self.migrated_tools_config not in tool_configs and os.path.exists(self.migrated_tools_config):
             tool_configs.append(self.migrated_tools_config)
         for path in tool_configs:
-            if not os.path.exists(path):
+            if not os.path.exists(path) and path != self.shed_tool_conf:
```

Comment:
@natefoo While trying to fix CI testing for BioBlend I noticed that:
- `path` may be a relative, e.g. `./config/shed_tool_conf.xml`, because it comes from `resolve_path(path, self.root)` in `_parse_config_file_options()` , where `self.root` is `.`
- `self.shed_tool_conf` is an absolute path, e.g. `/srv/galaxy/config/shed_tool_conf.xml`, because it comes from `resolve_path(kwargs.get('shed_tool_conf', 'shed_tool_conf.xml'), self.mutable_config_dir)` where `self.mutable_config_dir` is an absolute path.

Making `self.root` absolute seems a good way to solve this:
```diff
diff --git a/lib/galaxy/config/__init__.py b/lib/galaxy/config/__init__.py
index 92ab49fccd..aa1edd1f5f 100644
--- a/lib/galaxy/config/__init__.py
+++ b/lib/galaxy/config/__init__.py
@@ -100,7 +100,7 @@ def resolve_path(path, root):
 
 
 def find_root(kwargs):
-    root = kwargs.get('root_dir', '.')
+    root = os.path.abspath(kwargs.get('root_dir', '.'))
     return root
 
 
```

What do you think?